cmd/sync/main.go

```diff
@@ -81,33 +84,57 @@ func main() {
 	signal.Notify(sigterm, syscall.SIGTERM)
 
 	for {
-		for _, ups := range cfg.Upstreams {
-			ips, err := awsClient.GetPrivateIPsOfInstancesOfAutoscalingGroup(ups.AutoscalingGroup)
+		for _, upstream := range cfg.Upstreams {
+			ips, err := awsClient.GetPrivateIPsOfInstancesOfAutoscalingGroup(upstream.AutoscalingGroup)
 			if err != nil {
-				log.Printf("Couldn't get the IP addresses of instances of the Auto Scaling group %v: %v", ups.AutoscalingGroup, err)
+				log.Printf("Couldn't get the IP addresses of instances of the Auto Scaling group %v: %v", upstream.AutoscalingGroup, err)
 				continue
 			}
 
 			var backends []string
 			for _, ip := range ips {
-				backend := fmt.Sprintf("%v:%v", ip, ups.Port)
+				backend := fmt.Sprintf("%v:%v", ip, upstream.Port)
 				backends = append(backends, backend)
 			}
 
-			var added, removed []string
-
-			if ups.Kind == "http" {
-				added, removed, err = nginx.UpdateHTTPServers(ups.Name, backends)
+			if upstream.Kind == "http" {
+				var upsServers []UpstreamServer
+				for _, server := range backends {
```

Comment:
instead of iterating over backends, could we iterate over `ips`. same for line 120. this way we can remove lines 94-98.